File: jbback/src/main/java/org/zmalchunz/configurer/jbback/repositories/TemplateRepository.java
package org.zmalchunz.configurer.jbback.repositories;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;
import org.zmalchunz.configurer.jbback.api.Template;

import java.util.UUID;

@Repository
public interface TemplateRepository extends MongoRepository<Template, UUID> {
}

File: backend/src/main/kotlin/org/zmalchunz/configurer/backend/repositories/TemplateRepository.kt
package org.zmalchunz.configurer.backend.repositories

import org.springframework.stereotype.Service
import org.zmalchunz.configurer.backend.model.Template

@Service
class TemplateRepository {

    val repository: MutableList<Template> = ArrayList()

    fun findTemplates(): List<Template> = repository

    fun saveTemplate(template: Template): String {
        repository.add(template)
        return template.id
    }
}
